projects/media-viewer/src/lib/media-viewer/toolbar/search-bar/search-bar.component.ts

```diff
@@ -10,21 +10,59 @@ import { BehaviorSubject } from 'rxjs';
 export class SearchBarComponent {
 
   @Input() searchbarHidden: BehaviorSubject<boolean>;
-  @Input() actionEvents: ActionEvents;
+  @Input() searchEvents: Subject<SearchOperation>;
+  @Input() searchResultsCount: Observable<SearchResultsCount>;
 
-  private lastSearchTerm: string;
+  highlightAll = true;
+  matchCase = false;
+  wholeWord = false;
+  resultsText = '';
+  searchText = '';
+  haveResults = false;
 
   constructor() {}
 
-  searchNext(searchTerm: string) {
-    this.actionEvents.search.next(new SearchOperation(searchTerm, false, searchTerm !== this.lastSearchTerm));
+  ngOnInit() {
+    this.searchResultsCount.subscribe(r => this.onUpdateResults(r));
```

Comment:
it might be easier to use the async pipe to avoid having to subscribe to this. then we can just have a setter method that reacts to new values coming in, in the same way onUpdateResults is doing that

Reply:
I tried to do that but failed, for some reason when I used set it sent in the observable rather than the operation, I was muchos confused.